src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_DesignerAttributes.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.DesignerAttributes;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using RoslynLogger = Microsoft.CodeAnalysis.Internal.Log.Logger;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    // root level service for all Roslyn services
+    internal partial class CodeAnalysisService : IRemoteDesignerAttributeService
+    {
+        /// <summary>
+        /// This is top level entry point for DesignerAttribute service from client (VS).
+        /// 
+        /// This will be called by ServiceHub/JsonRpc framework
+        /// </summary>
+        public async Task<DesignerAttributeResult> ScanDesignerAttributesAsync(DocumentId documentId)
+        {
+            using (RoslynLogger.LogBlock(FunctionId.CodeAnalysisService_GetTodoCommentsAsync, documentId.ProjectId.DebugName, CancellationToken))
+            {
+                try
+                {
+                    var solution = await GetSolutionAsync().ConfigureAwait(false);
+                    var document = solution.GetDocument(documentId);
+
+                    var service = document.GetLanguageService<IDesignerAttributeService>();
```

Comment:
i'm a little confused by this.  Won't this then get the service that then will try to go talk to the remote host again?